pkg/trace/event/processor.go

```diff
@@ -83,14 +76,20 @@ func (p *Processor) Process(root *pb.Span, t pb.Trace) (events []*pb.Span, numEx
 		sampler.SetClientRate(span, clientSampleRate)
 		sampler.SetPreSampleRate(span, preSampleRate)
 		sampler.SetEventExtractionRate(span, extractionRate)
-		if hasPriority {
-			sampler.SetSamplingPriority(span, priority)
+		sampler.SetAnalyzedSpan(span)
+		if t.DroppedTrace {
+			events = append(events, span)
 		}
+		numEvents++
+	}
 
-		events = append(events, span)
+	// If `DroppedTrace` is false, we will just mark spans as analyzed in-place.
+	// If `DroppedTrace` is true, we will store analyzed spans in `events`, then replaces chunk's `Spans` with it.
+	if t.DroppedTrace {
```

Comment:
```suggestion
	if t.DroppedTrace {
		// we are not keeping anything out of this trace, except the events
```
The general style is that any explanation of what an `if` statement does, goes _inside_ the `if` statement. The reasoning is that otherwise no explanation is needed, and the user can skip reading it.

You may also notice that this is a very short sentence which makes sense and more clearly explains the same thing you did above in two lines instead of one. The user can see that this trace is dropped (due to the `DroppedTrace` field being `false`) and he can also see that the spans are replaced. The only thing missing is the explanation.

So, please remove the upper two lines of comment and just leave this one, unless you have a different opinion.